spicy/runtime/src/driver.cc

```diff
@@ -334,12 +338,12 @@ Result<hilti::rt::Nothing> Driver::processPreBatchedInput(std::istream& in) {
             else
                 return hilti::rt::result::Error(hilti::rt::fmt("unknown session type '%s'", m[2]));
 
-            if ( auto parser = lookupParser(parser_name) ) {
+            if ( auto parser = lookupParser(parser_name); parser && (*parser)->parse1 ) {
```

Comment:
I see that `_process` has similar uses of `parse1` which according to the updated logic can also be unset. We should add code there to check it before use as we do for `parse3` here now.

Reply:
Ack, missed protecting the  actual parse function. Just pushed a better fix.